Zend/zend_exceptions_arginfo.h

```diff
@@ -179,3 +179,143 @@ static const zend_function_entry class_ArithmeticError_methods[] = {
 static const zend_function_entry class_DivisionByZeroError_methods[] = {
 	ZEND_FE_END
 };
+
+
+static const zend_function_entry class_UnhandledMatchError_methods[] = {
+	ZEND_FE_END
+};
+
+#define register_class_Throwable(class_entry, class_entry_Stringable) \
+{ \
+	zend_class_entry ce; \
+ \
+	INIT_CLASS_ENTRY(ce, "Throwable", class_Throwable_methods); \
+	class_entry = zend_register_internal_interface(&ce); \
+}
```

Comment:
What's the reason for generating this as a macro rather than function?

Reply:
It mostly works... Unfortunately, I can't use the generated code for `Error` and `Exception`. If I did so, I would get the previously mentioned segfault. I have no idea why. The problem seems to be somewhat related to the `create_object`  registation and `zend_class_implements()`, because I get green tests when I use the below  code (+ I commented out the `zend_class_implements()` call in `register_class_Exception()`):

```
zend_ce_exception = register_class_Exception(zend_ce_throwable);
zend_ce_exception->create_object = zend_default_exception_new;
zend_class_implements(zend_ce_exception, 1, zend_ce_throwable);
```

I tried to pass `zend_default_exception_new` to `register_class_Exception()` and register it there, but it also failed.

Reply:
The problem here was the interaction with zend_implement_throwable, which required zend_ce_exception to be initialized early. I've now addressed this with some explicit checks.

Reply:
Thank you! It's very kind from you :)